### PR TITLE
62/implementation guide

### PR DESCRIPTION
**Pull request by spruce-bruce**
#62

### CODE REVIEW

File: docs/implementation.md [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+## Creating Users
+
+Methods for creating users is not defined by the OpenID Connect specification. All user creation is done by the OIDC Platform.
+
+### Inviting Users
+
+The OIDC Platform provides an endpoint that will invite your user to create an account. The invite workflow looks like this:
+
+1. Your application POSTs to the invite endpoing with your new users email address (plus some more information needed to get this user back to your application)
```
Comment (chrisshiplet) [inline docs/implementation.md]: endpoing

File: docs/implementation.md [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+
+##### response_types
+
+When a client application redirects their user to the login screen the client applictation will include a "response_type" string in the query parameters. The client for that application must be configured explicitly to allow that response_type. Response types can be `code`, `id_token`, `token` OR a [Multiple-Valued Response Type](http://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#Combination).
+
+##### grant_types
+
+A good guide for OAuth 2.0 grant types can be found [here](https://alexbilbie.com/guide-to-oauth-2-grants/). That guide defines grants like this
+
+> The OAuth 2.0 specification is a flexibile authorization framework that describes a number of grants (“methods”) for a client application to acquire an access token (which represents a user’s permission for the client to access their data) which can be used to authenticate a request to an API endpoint.
```
Comment (chrisshiplet) [inline docs/implementation.md]: flexibile

File: docs/implementation.md [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+
+All the client metadata can be found documented more completely in the OIDC spec [here](http://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata). Here is a brief explanation of the most important ones that hopefully will keep you from having to dig around too much in the spec.
+
+##### redirect_uris
+
+This is a JSON array of uris that the OIDC Platform will be allowed to return your users to. Your app will tell the OIDC Provider which of these uris to use when redirecting users back to your app after login. If it doesn't match one of these values _exactly_ then your user will not be allowed to login at all.
+
+##### response_types
+
+When a client application redirects their user to the login screen the client applictation will include a "response_type" string in the query parameters. The client for that application must be configured explicitly to allow that response_type. Response types can be `code`, `id_token`, `token` OR a [Multiple-Valued Response Type](http://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#Combination).
```
Comment (chrisshiplet) [inline docs/implementation.md]: applictation

**Comment by spruce-bruce** [comment]
Added a note about best practice for the client credentials grant

**Pull request merged**